### PR TITLE
[V3 Bank] Raise TypeError when passing a non-int transaction amount

### LINKED ISSUE #1975: [V3 Bank] bank.withdraw/deposit_credits breaks mongodb with floats
# Other bugs

If you give bank.deposit_credits or bank.withdraw_credit a float instead of an int while using MongoDB, the users balance is stuck as a double. Because of this many cogs break because they expect an int, but get a float.

#### What were you trying to do?

I was trying to withdraw credits from a users account with

`await bank.withdraw_credits(ctx.author, X * 0.30)`

although due to a bug I accidentaly ran

```
await bank.withdraw_credits(ctx.author, X * 0.30)
await bank.deposit_credits(ctx.author, X * 0.30)
````

This is most likely unrelated to the bug, but I thought I should mention it anyway.

#### What were you expecting to happen?

I expected red to either throw an error or convert the float to an int 

#### What actually happened?

Red carried on, as if the value was an int. It changed the value type of the balance in mongodb from an int32 to a double. This breaks many other cogs because they expect an int but get a double.

#### How can we reproduce this issue?

1. run the following command

```
await bank.withdraw_credits(ctx.author, X * 0.30)
await bank.deposit_credits(ctx.author, X * 0.30)
```

2. done

### PR DESCRIPTION
**Pull request by Tobotimus**
Resolves #1975

Non-int transaction amounts don't really make any logical sense, so instead of naïvely putting them into Config, we should make sure the user is supplying the correct type.

### CODE REVIEW

File: redbot/core/bank.py [hunk 1]
```diff
@@ -214,8 +214,12 @@ def _invalid_amount(amount: int) -> bool:
     ValueError
         If the withdrawal amount is invalid or if the account has insufficient
         funds.
+    TypeError
+        If the withdrawal amount is not an `int`.
 
     """
+    if not isinstance(amount, int):
+        raise TypeError("Withdrawal amount must be of type int, not {}".format(type(amount)))
```
Comment (Redjumpman) [inline redbot/core/bank.py]: Missing a period at the end of this statement.

File: redbot/core/bank.py [hunk 2]
```diff
@@ -245,8 +249,12 @@ def _invalid_amount(amount: int) -> bool:
     ------
     ValueError
         If the deposit amount is invalid.
+    TypeError
+        If the deposit amount is not an `int`.
 
     """
+    if not isinstance(amount, int):
+        raise TypeError("Deposit amount must be of type int, not {}".format(type(amount)))
```
Comment (Redjumpman) [inline redbot/core/bank.py]: Missing a period at the end of this statement.

File: redbot/core/bank.py [hunk 3]
```diff
@@ -269,14 +277,18 @@ def _invalid_amount(amount: int) -> bool:
...
     Raises
     ------
     ValueError
         If the amount is invalid or if ``from_`` has insufficient funds.
+    TypeError
+        If the amount is not an `int`.
 
     """
+    if not isinstance(amount, int):
+        raise TypeError("Transfer amount must be of type int, not {}".format(type(amount)))
```
Comment (Redjumpman) [inline redbot/core/bank.py]: Missing a period at the end of this statement.

**Comment by Redjumpman** [comment]
I marked a few lines for the type error being raised, where it was missing a period. Overall this looks pretty good. I think having it raise a TypeError versus changing floats to an int internally is a better way to go. An internal conversion could round the number to a value the user was not expecting.

To remark on the previous issue that was started, floats do not really make sense for economy, and it really causes some weird issues with the leader board when you get some crazy long float.

Other than the punctuation, this PR is ready.

**Note:** Only adding my supporting rationale for this PR in case this is revisited in the future.

**Comment by Tobotimus** [comment]
Thanks @Redjumpman, the requested changes have been made.

**Review by Redjumpman — APPROVED** [review]
This is ready to go for B19

**Pull request merged**